component-library/component-lib/src/shared/directives/dom-change.directive.ts

```diff
@@ -0,0 +1,33 @@
+import { Directive, ElementRef, EventEmitter, OnDestroy, Output } from '@angular/core';
```

Comment:
This would be a good directive to genericise to be used by other components.  It would replace a things like: @HostListener('window:resize', ['$event']). 



If we do go the generic route a LIFO stack could be implemented to make the directive more efficient ( to prevent lots of repainting on window resize) : 

const queue = [];const mo = new MutationObserver(mutations => {
  if (!queue.length) requestAnimationFrame(process);
  queue.push(mutations);
});function process() {
  for (const mutations of queue) {
    // ..........
}
queue.length = 0;
}